### PR TITLE
rbd: fix memory leak when fails to get the address

### PR DESCRIPTION
**Pull request by lxbsz**
Signed-off-by: Xiubo Li <xiubli@redhat.com>

### CODE REVIEW

File: rbd.c [hunk 1]
```diff
@@ -1159,9 +1159,11 @@ static int tcmu_rbd_open(struct tcmu_device *dev, bool reopen)
 #ifdef LIBRADOS_SUPPORTS_GETADDRS
 	/* Get current entry address for the image */
 	ret = rados_getaddrs(state->cluster, &state->addrs);
-	tcmu_dev_info(dev, "address: {%s}\n", state->addrs);
-	if (ret < 0)
-		return ret;
+	if (ret < 0) {
+		tcmu_dev_err(dev, "failed getting the address\n");
```
Comment (idryomov) [inline rbd.c:1163]: Nit: for consistency with other error messages in this function
```suggestion
		tcmu_dev_err(dev, "Could not get address.\n");
```

File: rbd.c [hunk 2]
```diff
@@ -1159,9 +1159,11 @@ static int tcmu_rbd_open(struct tcmu_device *dev, bool reopen)
...
+		goto stop_image;
+	}
+	tcmu_dev_info(dev, "opened, the address: {%s}\n", state->addrs);
```
Comment (idryomov) [inline rbd.c:1166]: Nit: keep the old string as the image is opened unconditionally whereas this message is printed only if `rados_getaddrs()` is supported (nautilus and later).
```suggestion
	tcmu_dev_info(dev, "address: {%s}\n", state->addrs);
```

File: rbd.c (hunk 1 above)
Comment (lxbsz) [reply rbd.c]: Will fix it.

File: rbd.c (hunk 2 above)
Comment (lxbsz) [reply rbd.c]: Will fix it.

**Comment by lxbsz** [comment]
Closed this one by accident, and have update it in a new PR https://github.com/open-iscsi/tcmu-runner/pull/675.

Thanks @idryomov

**Pull request closed without merging**